packages/squiggle-lang/src/XYShape.ts

```diff
@@ -81,6 +81,9 @@ const interpolate = (
   yMax: number,
   xIntended: number
 ): number => {
+  if (yMin === yMax) {
+    return yMin;
+  }
```

Comment:
This is important; `yMin * minProportion + yMax * maxProportion` was precise with `Math.random()` values which ended with `0` bit, but not with other RNGs.